### DISCUSSION TITLE
Training data size and event distribution for LMU time series

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#1065] [General Discussion]
> _Archived from the Nengo Discourse forum._ Originally posted by **fladventurerob** on 2022-01-08. (Forum topic #2025, post #1 / id 6195)

Hello all, I have done quite a bit of research, and read the paper numerous times, however I haven't been able to get a clear understanding that can provide an answer to my below questions. Thank you in advance for any help and clarity you can offer! 

I have a few questions, as follows:

[Question 1] I understand the LMU to function similar to a standard RNN, as it relates to time series data. Does a sliding window work with the LMU for event based data?

[Question 2] Does unbalanced data have an effect on the LMU?

[2a] Is small batch training something compatible with the LMU?

[Question 3] Should the "size" of the LMU be determined by the total amount of training data, or, rather, the size of each subset of data (i.e., each individual timeseries). 

Thank you very much in advance, and if you have any questions or need clarification, please let me know.

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **Eric** on 2022-01-11. (Forum topic #2025, post #2 / id 6211)

These sound like research questions to me! I don't know of any existing applications of LMUs to event-based data of the type that you're describing, so I think these are open questions.

To answer them, I would start by creating a simple LMU memory system (that is, just the dynamical system defined by the A and B matrices), and seeing how well it can encode/decode some of your event-based data. Here's a notebook that looks at the ability of an LMU memory to encode some simple signals:
[lmu-basics.ipynb|attachment](https://raw.githubusercontent.com/nengo/forum-archive/main/uploads/default/original/2X/f/f37330d6f94b262037c5b2dcb38d8b6b3742fc55.ipynb) (676.5 KB)

In terms of training with unbalanced data, I think the LMU is not special in this regard, since it's simply a type of neural network layer, and the problems of unbalanced data have more to do with the training process (specifically the loss function). If you have unbalanced data, I would recommend using a weighting scheme that weights rarer examples more highly in your loss function, or presenting examples in a more balanced manner. That said, I can't comment on your specific problem, and it may be possible to get reasonable results without accounting for the unbalanced data in this way.

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **fladventurerob** on 2022-01-13. (Forum topic #2025, post #3 / id 6220)

Haha, I had a feeling I would get an answer along those lines @​Eric but I really appreciate the information you provided, as well as the notebook. Sounds like I have some work/research to do!
